rust-src/key_derivation/src/lib.rs

```diff
@@ -0,0 +1,256 @@
+use ed25519_hd_key_derivation::derive;
+use keygen_bls::keygen_bls;
+use pairing::bls12_381::FrRepr;
+use serde::Deserialize;
+use std::fmt;
+
+#[derive(Deserialize)]
+pub enum Net {
+    Mainnet,
+    Testnet,
+}
+
+impl fmt::Display for Net {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Net::Mainnet => write!(f, "919"),
+            Net::Testnet => write!(f, "1"),
+        }
+    }
+}
+
+fn bls_key_bytes_from_seed(key_seed: [u8; 32]) -> [u8; 32] {
+    let bls_key_fr = keygen_bls(&key_seed, b"").unwrap();
+    let fr_repr = FrRepr::from(bls_key_fr);
+    let fr_repr_ref = fr_repr.as_ref();
+    let key_bytes: Vec<u8> = fr_repr_ref
+        .iter()
+        .rev()
+        .flat_map(|val| val.to_be_bytes())
+        .collect();
+    let mut bls_key = [0u8; 32];
+    bls_key.clone_from_slice(&key_bytes[0..32]);
+    bls_key
+}
+
+pub struct ConcordiumHdWallet {
+    pub seed: Vec<u8>,
```

Comment:
We may want to impose a restriction that this only allows for a 24 word seed phrase, i.e. for 64 bytes instead of it being adjustable.

Reply:
Is there a reason to be more general at this point? Otherwise I would prefer that this is more strict.

Additionally, this is part of the public API so should be well-documented. It is also useful to have Debug implementation however in this case it should be manual to not print the seed I imagine?

Reply:
I think in principle, people could make their own wallet with 12 word seed phrases. It's less secure, so we don't do it in our wallets, but it should still work...

Reply:
I have made the restriction that is must be 64 bytes.

Reply:
I think it would be fine to have a Debug implementation that also prints the seed. I suppose being in Debug mode is a controlled environment so I think it would be okay.